### PR TITLE
Tensor reinitialization codemod

### PR DESCRIPTION
**Pull request by jerryzh168**
Summary:
Codemod generated with clangr shard mode, 25 files per diff,
To eliminiate partially initialized Tensor, we split the initialization of local Tensor variables into two steps, first declare un uninitialized Tensor, and
call `ReinitializeTensor` to initialize it.
motivation: https://github.com/pytorch/pytorch/pull/12407

Differential Revision: D13717360

### CODE REVIEW

File: caffe2/experiments/operators/fully_connected_op_decomposition.h [hunk 1]
```diff
@@ -67,7 +67,7 @@ class FullyConnectedOpDecomp final : public Operator<Context> {
     std::vector<int64_t> dims;
     if (X.dim() > 1) {
       dims = {M, N};
-      multi_buffer_.Resize(M, middle);
+      ReinitializeTensor(&multi_buffer_, {M, middle}, at::dtype<T>().device(Context::GetDeviceType()));
```
Comment (ezyang) [inline caffe2/experiments/operators/fully_connected_op_decomposition.h]: `.device(Context::GetDeviceType()` is unnecessary right?

**Comment by ezyang** [comment]
Is there a reason you're not using ghexport to export these diffs? I just realized that this one is stacked (I was wondering why it seemed really long, longer than 25 files)

File: caffe2/experiments/operators/fully_connected_op_decomposition.h (hunk 1 above)
Comment (jerryzh168) [reply caffe2/experiments/operators/fully_connected_op_decomposition.h]: `ReinitializeTensor` is a free function, we need to provide device

**Comment by jerryzh168** [comment]
Hmm... this is not meant to be stacked on other diffs in the beginning, so I used the normal export which is more convenient.

File: caffe2/experiments/operators/fully_connected_op_prune.h [hunk 2]
```diff
@@ -349,9 +348,9 @@ namespace caffe2 {
         }
 
       protected:
-       Tensor bias_multiplier_{Context::GetDeviceType()};
+       Tensor bias_multiplier_;
        Tensor sum_buffer_{Context::GetDeviceType()};
```
[the remark this answers is not in the document]
Comment (jerryzh168) [reply caffe2/experiments/operators/fully_connected_op_prune.h]: modified, this is not captured by codemod because it is using ResizeLike

**Pull request closed without merging**